### PR TITLE
verify config key shares are valid

### LINKED ISSUE #432: Verify that our key shares fit to the pub key sets
I just did some manual config migration and made my signet federation very upset by giving it a LN threshold decryption key share that didn't fit the pub key. Since I'm running a monomint right now there is no peer to ban, so it just complains that it made a faulty contribution each epoch and starts the next in the hope that it will get better (it won't).

Having #331 would be really helpful in rectifying the situation by deleting/editing a DB entry …

### PR DESCRIPTION
**Pull request by jkitman**
Fixes #432

### CODE REVIEW

**Comment by justinmoon** [comment]
LGTM. Maybe it should print out more informative error messages than just "thread 'main' panicked at 'assertion failed: `(left == right)`"?

**Review by justinmoon — APPROVED** [review]
LGTM

File: fedimint-api/src/config.rs [hunk 1]
```diff
@@ -17,4 +17,7 @@ pub trait GenerateConfig: Sized {
     ) -> (BTreeMap<PeerId, Self>, Self::ClientConfig);
 
     fn to_client_config(&self) -> Self::ClientConfig;
+
+    /// Asserts that the public keys in the config are and panics otherwise (no way to recover)
+    fn validate_config(&self, identity: &PeerId);
```
Comment (elsirion) [inline fedimint-api/src/config.rs:22]: Eventually we want errors here imo to have more readable output for users. But right now that's too much work. 80/20

File: fedimint/src/config.rs [hunk 2]
```diff
@@ -156,6 +156,33 @@ impl GenerateConfig for ServerConfig {
...
+            self.epoch_pk_set.public_key_share(identity.to_usize()),
+            "Epoch private key doesn't match pubkey share"
+        );
+        assert_eq!(
+            self.hbbft_sks.public_key_share(),
+            self.hbbft_pk_set.public_key_share(identity.to_usize()),
+            "HBBFT private key doesn't match pubkey share"
+        );
+        assert_eq!(
+            self.peers.keys().max().copied().map(|id| id.to_usize()),
```
Comment (elsirion) [inline fedimint/src/config.rs:172]: I think at some point we should try to fix this assumption

**Pull request merged**